### PR TITLE
System.text.json support

### PR DESCRIPTION
**Pull request by andrueastman**
This PR is part of https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/issues/80

### CODE REVIEW

File: src/Microsoft.Graph/Generated/callrecords/model/CallRecord.cs [hunk 1]
```diff
@@ -12,13 +12,12 @@ namespace Microsoft.Graph.CallRecords
...
     using System.IO;
-    using System.Runtime.Serialization;
-    using Newtonsoft.Json;
+    using System.Text.Json.Serialization;
 
     /// <summary>
     /// The type Call Record.
     /// </summary>
-    [JsonObject(MemberSerialization = MemberSerialization.OptIn)]
+
```
Comment (MIchaelMainer) [inline src/Microsoft.Graph/Generated/callrecords/model/CallRecord.cs:20]: Generating a whitespace character. When ignoring the the attributes and using statement changes, I still see this with:
`git diff origin/feature/2.0 -I "\s{1,}\[Json.*" -I "\s{1,}using.*" > diff.txt`

If I run the following to now ignore the space charactoer, there is only a single change around the lack of no newline in exceltests:
`git diff origin/feature/2.0 -I "\s{1,}\[Json.*" -I "\s{1,}using.*" -I "\s" > diff.txt`

**Review by MIchaelMainer — APPROVED** [review]
Update the template so we can remove the whitespace change, and then I will approve again.
Provide links to the PR used to make these changes.

**Comment by andrueastman** [comment]
@MIchaelMainer Thanks for taking a look at this! This is much appreciated
I also have a task to merge in generator updates with the composable functions work then I will move this out of draft state.

File: src/Microsoft.Graph/Generated/callrecords/model/CallRecord.cs (hunk 1 above)
Comment (andrueastman) [reply src/Microsoft.Graph/Generated/callrecords/model/CallRecord.cs:20]: Thanks!

**Review by MIchaelMainer — APPROVED** [review]
Sample reviewed files locally. Shows 6582 changed files as stated locally. For both v1 and beta, I noticed that in ModelSerializationTests, the generic DirectoryObject parameter was changed to Entity. I'm curious as why this change occurred. 

On another note, we should consider removing the generation of constructors that set odata.type. It is a cause of customer pain. Before we do that, we'll want to take inventory of where it is required. We can talk about this later.

**Comment by andrueastman** [comment]
> I noticed that in ModelSerializationTests, the generic DirectoryObject parameter was changed to Entity.

For some reason the [DirectoryObject](https://github.com/microsoftgraph/msgraph-beta-sdk-dotnet/blob/andrueastman/system.text.json/src/Microsoft.Graph/Generated/model/Entity.cs) class doesn't seem to get the **DerivedTypeCoverter** annotation like the [Entity](https://github.com/microsoftgraph/msgraph-beta-sdk-dotnet/blob/andrueastman/system.text.json/src/Microsoft.Graph/Generated/model/Entity.cs) class.
I suspect this is related to https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/issues/237

**Comment by andrueastman** [comment]
https://github.com/microsoftgraph/msgraph-beta-sdk-dotnet/pull/209/commits/c6c5ba8dd54f2692b8703618a7e07563c5929c19 Resolves indentation issues

**Comment by MIchaelMainer** [comment]
I suspect that DirectoryObject doesn't get it since it isn't abstract. I don't think it is an issue.

**Comment by andrueastman** [comment]
Thanks for clearing that up for me. 👍

**Pull request merged**